torchao/experimental/quant_api.py

```diff
@@ -628,15 +621,28 @@ def apply(weight, bias: Optional[torch.Tensor] = None):
                 assert (
                     TORCH_VERSION_AT_LEAST_2_6
                 ), "aten target is requires torch version > 2.6.0"
+                # Fallback to Channelwise scheme if group_size is too big
+                if weight.shape[-1] < group_size:
+                    logger.warning(f"Changing group_size to {
+                                   weight.shape[-1]}. Weight shape {weight.shape} can not support group_size {group_size}.")
+                    group_size = weight.shape[-1]
                 if torch.backends.kleidiai.is_available():
-                    if isinstance(granularity, PerGroup):
+                    if weight.shape[-1] != group_size and group_size % 32 == 0:
```

Comment:
`weight.shape[-1] != group_size` I don't understand this constraint.

IIRC the constraint is `k % group_size == 0 and group_size % 32 == 0`